### DISCUSSION TITLE
Help us translate Xplorer and its Docs!

### DISCUSSION THREAD

**Discussion by kimlimjustin** [kimlimjustin/xplorer#30] [General]
## Help us translate Xplorer and its Docs!
The Xplorer i18n support is ready, and it's time for Xplorer and its docs to be translated.

## Translation Process
We host our locales using the [crowdin](https://crowdin.com) to help the process of translation. To translate it, please follow these steps:
* SIgn up on [Crowdin](https://crowdin.com) and Join our project [here](https://crwd.in/xplorer).
* Make sure your locale exists there, if it does not exist, leave a comment down and I'll add the language option :)
* Get familiar with the Crowdin translation UI, as you will need to use it to translate JSON and Markdown files
* Translate the content!

## Priority Files to translate on Crowdin
1. `src/Locales` files
2. `docs/docs` files
3. `docs/i18n/en` files
4. `docs/community` files

## Production
Once the files on `src/Locales` have been translated for more than 80%, we will add it into Xplorer app, and for the docs, we will add it into production once the translation looks good!

Please comment down if you have any questions!

---

Thanks for your help!

**Comment by UshiHiraga** [comment]
Hello. I want to help to translate it to spanish, but the locale doesn't exist...

**Reply by kimlimjustin** [reply]
Thanks for your interest! Added Spanish!

**Comment by silas229** [comment]
# Add German language
I‘ve already created a pr #51, but it would be easier to add German on Crowdin for additional translation

**Reply by kimlimjustin** [reply]
Thanks, Added German on Crowdin! Please copy the changes of the PR to Crowdin too, (I can do it if you are lazy to haha)

**Comment by lumiuko** [comment]
I'd like to help with translation into Russian

**Reply by kimlimjustin** [reply]
Thanks for your interest! Added the Russian language into Crowdin.

**Comment by yay99857** [comment]
I'd like to help with translation into Portuguese

**Reply by kimlimjustin** [reply]
Thanks for your interest! Added the Portuguese into Crowdin.

**Reply by yay99857** [reply]
I forgot to mention, is Portuguese from Brazil, idk if Crowdin has this difference.

**Reply by kimlimjustin** [reply]
Oh okay, updated!

**Comment by ukjinjang** [comment]
I found Xplorer at dev.to and I want to help with Korean translation! :)

**Reply by kimlimjustin** [reply]
Thanks for your interest! Added the Korean into Crowdin.

**Comment by KayVeeCodes** [comment]
I came from dev.to and I am amazed with the app. I would like to offer translation in Greek

**Reply by kimlimjustin** [reply]
Thanks for your intereset! Added the Greek into Crowdin :)

**Comment by YoussefRabeiii** [comment]
I came from dev.to and i would love to help with translation into Arabic
 and code if you need any help

**Reply by kimlimjustin** [reply]
Thanks for your interest! Added Arabic into Crowdin :)

**Comment by thomasbnt** [comment]
Hello ! I would like to contribute to add French language ;)

**Reply by kimlimjustin** [reply]
Thanks for your interest! Added French into Crowdin :)

**Comment by javaboy-github** [comment]
I'm Japanese and I want to translate to Japanese.
(This repository has some bad translations.)
I'm 13 years old. I can't sign up on Crowdin:cry: Can I directly fork this repository and send pr?

**Reply by kimlimjustin** [reply]
Ah, okay then, but please study the project structure, you can ask if you got something you do not understand.

Also, there's something I translated myself, you can edit it if you feel there's another phrase better :)

Thanks!

**Reply by javaboy-github** [reply]
I have a question.
Where can I create a Japanese version of a file in docs/docs a
or docs/community?

**Reply by kimlimjustin** [reply]
This is quite complicated since you have to follow the [docusaurus docs](https://docusaurus.io/docs/i18n/git#initialize-the-i18n-folder).

Or do you have any contact method (like email)? So that I can send you the structures of Japanese locales.

Thanks!

**Reply by javaboy-github** [reply]
Thanks!
I'll try to read the document.

**Reply by comradekingu** [reply]
@kimlimjustin
The reason people under 18 can't use Crowdin is it necessitates signing a legal contract.
You are forwarding this work as per the terms and conditions if it makes it into Crowdin.
Happy to help if you use Weblate, which needs no such small print.

**Reply by kimlimjustin** [reply]
> @kimlimjustin The reason people under 18 can't use Crowdin is it necessitates signing a legal contract. You are forwarding this work as per the terms and conditions if it makes it into Crowdin. Happy to help if you use Weblate, which needs no such small print.

thanks for the suggestion, I'll take a look soon.

**Comment by misobarisic** [comment]
Hello,
I would like to translate this project to Croatian. I am a native speaker.

Thanks

**Reply by kimlimjustin** [reply]
Thanks for your interest! Added Croatian into Crowdin :) Also, please check if I added the correct one since it was the first time I heard Croatian,

**Reply by misobarisic** [reply]
Can confirm, you added the correct one. Cheers

**Comment by comradekingu** [comment]
Happy to help if you move away from Crowdin.

**Reply by comradekingu** [reply]
Just wanted people to know that is andrii-bodnar downvoting, a person that works at Crowdin.

**Comment by slayercio** [comment]


**Reply by kimlimjustin** [reply]
Thanks for your interest! Added polish into Crowdin :)

**Comment by NicolasGirondi** [comment]
Hello. I'd like to try translating Xplorer to Italian!

**Reply by kimlimjustin** [reply]
Thanks for your interest! Added Italian into Crowdin :)

**Reply by NicolasGirondi** [reply]
Hello. I just wanted to let you know that the Italian translation on Crowdin is 100% completed.

**Reply by stanleyowen** [reply]
Appreciate it really much👏🏻, thank you!

**Reply by NicolasGirondi** [reply]
Hello. I've noticed some errors in my italian translation, so i corrected it. When you want, you can update my changes. Thank you

**Comment by itsbahrami** [comment]
Hello. I would like to help this awesome project. I can help in Persian translation.

**Comment by AlidesChimin** [comment]
Hello! I have completed a Brazilian Portuguese translation for the current next branch, based on apps/client/src/locales/en.json. The old Crowdin invitation at crwd.in/xplorer now redirects to the Crowdin homepage. Is the Crowdin project still active and synchronized with the current repository? If so, could you provide the current project link and contributor access? Otherwise, would you prefer a focused pull request containing apps/client/src/locales/pt-BR.json and the necessary locale registration changes?
